File: app/src/main/java/com/hjc/reader/ui/wan/adapter/NavigationAdapter.java
package com.hjc.reader.ui.wan.adapter;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.reader.R;
import com.hjc.reader.bean.response.WanNavigationBean;
import com.hjc.reader.databinding.ItemNavigationBinding;


public class NavigationAdapter extends BaseQuickAdapter<WanNavigationBean.DataBean, BaseViewHolder> {
    private OnSelectListener listener;

    public NavigationAdapter() {
        super(R.layout.item_navigation);
    }

    @Override
    protected void onItemViewHolderCreated(@NonNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, WanNavigationBean.DataBean item) {
        if (item == null) {
            return;
        }

        ItemNavigationBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setNavigationBean(item);
            binding.tvChapter.setSelected(item.isSelected());

            if (listener != null) {
                binding.tvChapter.setOnClickListener(v -> {
                    listener.onSelected(helper.getAdapterPosition());
                });
            }
        }
    }

    public interface OnSelectListener {
        void onSelected(int position);
    }

    public void setOnSelectListener(OnSelectListener listener) {
        this.listener = listener;
    }
}

File: module-wan/src/main/java/com/hjc/learn/wan/adapter/NavigationAdapter.kt
package com.hjc.learn.wan.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.learn.wan.R
import com.hjc.learn.wan.databinding.WanItemNavigationBinding
import com.hjc.library_net.model.WanNavigationBean

/**
 * @Author: HJC
 * @Date: 2021/2/22 17:08
 * @Description: 导航数据侧边adapter
 */
class NavigationAdapter(data: MutableList<WanNavigationBean>?) :
    BaseQuickAdapter<WanNavigationBean, BaseViewHolder>(R.layout.wan_item_navigation, data) {

    private var listener: OnSelectListener? = null

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: WanNavigationBean) {
        val binding = DataBindingUtil.getBinding<WanItemNavigationBinding>(holder.itemView)
        binding?.let {
            it.navigationBean = item
            it.tvChapter.isSelected = item.isSelected
            it.tvChapter.setOnClickListener { listener?.onSelected(holder.adapterPosition) }
        }
    }

    interface OnSelectListener {
        fun onSelected(position: Int)
    }

    fun setOnSelectListener(listener: OnSelectListener?) {
        this.listener = listener
    }
}